geomstats/information_geometry/binomial.py

```diff
@@ -151,19 +149,24 @@ def pmf(k):
             -------
             pmf_at_k : array-like, shape=[..., n_points]
             """
-
-            def _pmf(param):
-                return binom.pmf(k, self.n_draws, param)
-
-            if n_batch:
-                return gs.array([_pmf(param) for param in point])
-
-            return gs.array(_pmf(point))
+            k = gs.reshape(gs.array(k), (-1,))
+            k_shape = gs.ones_like(k)
+            point_shape = gs.ones_like(point)
+            point_aux = gs.einsum("...i,j->...j", point, k_shape)
+            k_aux = gs.einsum("...i,j->...j", point_shape, k)
```

Comment:
why don't we use `gs.power` here? (I think we spoke about it at a very early stage, but I can't remember what was the drawback... maybe it was `tf`, which is not a problem anymore)

`gs.power(point, k)` seem to output the same as this for lines.

Reply:
I guess by using `gs.power` you mean not to broadcast but to use inputs as they are? Indeed, I just tested the version without broadcasting and it works well, I cannot remember either why I chose to broadcast.